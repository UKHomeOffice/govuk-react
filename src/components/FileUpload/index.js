// https://github.com/alphagov/govuk-frontend/tree/master/src/components/file-upload

import React from "react";
import PropTypes from "prop-types";
import glamorous from "glamorous";

import Label from "../Label/index";
import LabelText from "../LabelText/index";
import ErrorText from "../ErrorText/index";
import HintText from "../HintText/index";

import { FONT_SIZE, NTA_LIGHT, MEDIA_QUERIES } from "../../constants/index";

const Input = glamorous.input({
  boxSizing: "border-box",
  fontFamily: NTA_LIGHT,
  WebkitFontSmoothing: "antialiased",
  MozOsxFontSmoothing: "grayscale",
  fontWeight: 400,
  textTransform: "none",
  fontSize: FONT_SIZE.SIZE_14,
  width: "100%",
  padding: "5px 4px 4px",
  [MEDIA_QUERIES.LARGESCREEN]: {
    width: "50%"
  },
  "[disabled]": {
    cursor: "auto"
  }
});

const FileUpload = ({ meta, children, hint, acceptedFormats }) => (
  <Label error={meta.error}>
    <LabelText error={meta.error}>{children}</LabelText>
    {hint && <HintText>{hint}</HintText>}
    {meta.touched && meta.error && <ErrorText>{meta.error}</ErrorText>}
    <Input type="file" accept={acceptedFormats} error={meta.error} />
  </Label>
);

FileUpload.defaultProps = {
  hint: null,
  meta: {},
  acceptedFormats: null
};

FileUpload.propTypes = {
  hint: PropTypes.string,
  meta: PropTypes.shape({
    active: PropTypes.bool,
    dirty: PropTypes.bool,
    dirtySinceLastSubmit: PropTypes.bool,
    error: PropTypes.any,
    initial: PropTypes.bool,
    invalid: PropTypes.bool,
    pristine: PropTypes.bool,
    submitError: PropTypes.any,
    submitFailed: PropTypes.bool,
    submitSucceeded: PropTypes.bool,
    touched: PropTypes.bool,
    valid: PropTypes.bool,
    visited: PropTypes.bool
  }),
  children: PropTypes.node.isRequired,
  acceptedFormats: PropTypes.string
};

export default FileUpload;
