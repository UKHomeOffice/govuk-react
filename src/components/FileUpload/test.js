import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";
import FileUpload from "./";

describe("FileUpload", () => {
  const example = "example";
  const wrapper = <FileUpload>{example}</FileUpload>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(shallow(wrapper)).toMatchSnapshot(`enzyme.shallow`);
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
    expect(render(wrapper)).toMatchSnapshot(`enzyme.render`);
  });
});
