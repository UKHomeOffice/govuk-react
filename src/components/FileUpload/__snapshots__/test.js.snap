// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload matches snapshot: enzyme.mount 1`] = `
.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  width: 100%;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.glamor-2:after,
[data-glamor-2]:after {
  content: '';
  display: table;
  clear: both;
}

.glamor-0,
[data-glamor-0] {
  font-family: "nta", Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: block;
  clear: none;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  padding-bottom: 2px;
}

.glamor-1,
[data-glamor-1] {
  box-sizing: border-box;
  font-family: "nta", Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  width: 100%;
  padding: 5px 4px 4px;
}

@media only screen and (min-width: 641px) {
  .glamor-2,
  [data-glamor-2] {
    max-width: 960px;
  }
}

@media only screen and (min-width: 641px) {
  .glamor-0,
  [data-glamor-0] {
    font-size: 19px;
    line-height: 1.31579;
  }
}

@media only screen and (min-width: 641px) {
  .glamor-1,
  [data-glamor-1] {
    width: 50%;
  }
}

<FileUpload
  acceptedFormats={null}
  hint={null}
  meta={Object {}}
>
  <Label>
    <glamorous(label)>
      <label
        className="glamor-2"
      >
        <LabelText>
          <glamorous(span)>
            <span
              className="glamor-0"
            >
              example
            </span>
          </glamorous(span)>
        </LabelText>
        <glamorous(input)
          accept={null}
          type="file"
        >
          <input
            accept={null}
            className="glamor-1"
            type="file"
          />
        </glamorous(input)>
      </label>
    </glamorous(label)>
  </Label>
</FileUpload>
`;
